packages/flutter/lib/src/cupertino/checkbox.dart

```diff
@@ -149,6 +218,23 @@ class CupertinoCheckbox extends StatefulWidget {
 
   /// The color and width of the checkbox's border.
   ///
+  /// This property can be a [WidgetStateBorderSide] that can
+  /// specify different border color and widths depending on the
+  /// checkbox's state.
+  ///
+  /// Resolves in the following states:
+  ///  * [WidgetState.pressed].
+  ///  * [WidgetState.selected].
+  ///  * [WidgetState.hovered].
+  ///  * [WidgetState.focused].
+  ///  * [WidgetState.disabled].
+  ///  * [WidgetState.error].
+  ///
+  /// If this property is not a [WidgetStateBorderSide] and it is
+  /// non-null, then it is only rendered when the checkbox's value is
+  /// false. The difference in interpretation is for backwards
+  /// compatibility.
```

Comment:
Interesting, is that something we want/need to duplicate in CupertinoCheckbox?

Reply:
Yes. It allows developers to configure border properties in varying `WidgetState`s.